vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
I'm afraid that doesn't cut it either. [`Walkable`](https://sourcegraph.com/sourcegraph.com/sourcegraph/rwvfs@fe6b841a863a98172a6c5f03a65fabee6e9774ca/.tree/vfs.go#def=/sourcegraph.com/sourcegraph/rwvfs@fe6b841a863a98172a6c5f03a65fabee6e9774ca/.GoPackage/sourcegraph.com/sourcegraph/rwvfs/.def/Walkable&startbyte=2936&endbyte=3158) takes [`FileSystem`](https://sourcegraph.com/sourcegraph.com/sourcegraph/rwvfs@fe6b841a863a98172a6c5f03a65fabee6e9774ca/.tree/vfs.go#def=/sourcegraph.com/sourcegraph/rwvfs@fe6b841a863a98172a6c5f03a65fabee6e9774ca/.GoPackage/sourcegraph.com/sourcegraph/rwvfs/.def/FileSystem&startbyte=185&endbyte=580) as argument, which is not the same as `vfs.FileSystem`.